test/case/convention/openshift/golang-osd-operator/01-generated-files-checker

```diff
@@ -2,45 +2,33 @@
 
 set -e
 
-REPO_ROOT=$(git rev-parse --show-toplevel)
+REPO_ROOT="$(git rev-parse --show-toplevel)"
 
-source $REPO_ROOT/test/lib.sh
+source "${REPO_ROOT}/test/lib.sh"
 
 echo "Testing generators"
-repo=$(empty_repo)
-add_cleanup $repo
+repo="$(empty_repo)"
+add_cleanup "${repo}"
 test_project="file-generate"
 
-bootstrap_project $repo ${test_project} openshift/golang-osd-operator
-cd $repo
+# Attempt to boilerplate the test project with the openshift/golang-osd-operator convention
+bootstrap_project "${repo}" "${test_project}" openshift/golang-osd-operator
+cd "${repo}"
 make boilerplate-update
 go mod download
 export GOPATH=$GOPATH:`pwd`
 
-# Check failure when project is not committed
-make generate-check && err "make generate-check should have failed" || echo "Project is not committed. Checks failed as expected"
-
-# Checking files haven't been generated and the project is not committed
-make isclean || echo "Project is detected as uncommitted as expected"
-if [ -f src/test/test-generation ] || [ -f src/test/subfolder/test-generation-subfolder ] || [ -f pkg/apis/mygroup/v1alpha1/zz_generated.deepcopy.go ] ; then
-    err "Files have been generated while the project wasn't committed..."
-fi
-
-# Check failure when project is committed but generated content not generated/committed
-git add -A
-git commit -m "initial commit"
-
-make generate-check && err "make generate-check should have failed" || echo "Generated files are not committed. Checks failed as expected"
+# Generate files and check if the generated files matches the expected content
+make generate
```

Comment:
So... why can't we keep the `generate-check`s in here?

It's possible you'll need to update more of the files in src/test/ to match whatever the newer generators are putting out (like you did for file-generate).

Reply:
It's a mix of me making file-generate a better example project and thinking that at some point the behavior changed and this test wasn't even functioning as intended, e.g.

1. `make generate-check` only should fail as expected because the original file-generate operator didn't have its openapi file generated, so then this would always fail. In the new file-generate operator, I have the correct openapi file already in place.
2. `make generate-check` already includes `make isclean`, so the second check is redundant
3. None of those files: `src/test/test-generation`, `srcs/test/subfolder/test-generation-subfolder`, nor `pkg/apis/mygroup/v1alpha1/zz_generated.deepcopy.go` would ever exist - this must be a check based on older logic
4. Since `make generate-check` in step 1 calls `make generate`, and then any generated files get committed, it didn't make sense why this final `make generate-check` should fail.

So I decided to just generate the files, keep the existing simple diffs/existence checks, and keep the useful `make generate-check` tests afterwards.

Reply:
Yeah, looking at the rules and this test again, it didn't make a whole lot of sense. If you got it working, I'm happy :)